### PR TITLE
Adding E2E tests for my-posts page Part 1

### LINKED ISSUE #1169: E2E test - /my-posts page - Multiple issues available
## Context

We need to ensure that our core application features and user flows are fully covered by end-to-end (E2E) tests. We are using Playwright for these tests, and the workflow framework for E2E testing is already in place.

If you navigate to the e2e folder in the root of the project, you'll find that @JohnAllenTech has written numerous tests for the article page. These tests have been grouped into two categories: authenticated and unauthenticated. Please ensure that this pattern is followed when adding new tests. 

## Expected Behavior

Below is a list of possible tests for the `/my-posts` page. These are suggested tests generated with the help of ChatGPT, meant to provide a ***starting point***. You should use your discretion to refine these tests and account for relevant edge cases. 

Additionally, feel free to propose other valid tests that might apply specifically to the functionality of the '/my-posts' page.

Tests should be standalone and focus on a single purpose, ensuring each test verifies only one specific behavior or functionality.

Test should be added to e2e/my-posts.spec.ts

Please leave a comment specifying the tests you plan to write, so others don't duplicate the same efforts. 
All PRs are accepted for Hacktoberfest.

**Note:** For this page, the unauthenticated test is only necessary to verify that the user is correctly redirected to the `/get-started` page.

**Note:** As multiple people may contribute to this issue, please ensure you pull the latest changes from the upstream and resolve any conflicts. [Sync the upstream](https://www.codu.co/articles/effortless-upstream-syncing-a-custom-alias-for-git-users-fowfliwo)

## Tests

Here are some suggested E2E tests for the page based on your provided details:

### General Tests:
1. **Tab Navigation**
   - **Test:** Ensure the user can navigate between the "Drafts", "Scheduled", and "Published" tabs.
   - **Scenario:** Click on each tab and verify that the correct content is displayed.

### For the "Published" Tab:
2. **Display Published Date**
   - **Test:** Verify the correct display of "Published on" and "Last updated on" dates.
   - **Scenario:** For a post that has only been published, it should say "Published on [date]". If a post has been edited after publication, it should say "Last updated on [date]".

3. **Sorting by Published Date**
   - **Test:** Verify that posts in the "Published" tab are sorted correctly, with the most recently published or updated post appearing first. 
   - **Scenario:** Check that the order of posts reflects the latest publication or update date, with updated posts using the `updatedAt` field.
   **Note** Refer to issue #1095 by @HarshitT00 to see how the "Published posts" should be sorted. 
   ```In published posts section as the post is published update post is done so updatedAt is set to slightly greater time than published so use 1 minute time difference to distinguish between a published post and a post that published but updated later``` 

4. **Edit Published Post**
    - **Test:** Ensure the "Edit" option in the dropdown works for published posts.
    - **Scenario:** Select "Edit", make changes, and verify that the "Last updated on" timestamp changes correctly after saving.

5. **Delete Published Post**
    - **Test:** Verify the "Delete" option in the dropdown works for published posts.
    - **Scenario:** Select "Delete" and confirm that the post is removed from the list.

### For the "Drafts" Tab:
6. **Sorting by Last Updated**
   - **Test:** Verify that posts in the "Drafts" tab are sorted chronologically by their last updated date.
   - **Scenario:** Confirm that the first post in the list is the most recently updated, and the last post is the oldest.
   
7. **Edit Draft Post**
   - **Test:** Ensure the "Edit" option in the dropdown works for draft posts.
   - **Scenario:** Select "Edit" from the dropdown, make changes, and save. Verify that the post is updated and reflected correctly on the page.

8. **Delete Draft Post**
   - **Test:** Verify the "Delete" option in the dropdown works for draft posts.
   - **Scenario:** Select "Delete" from the dropdown and confirm deletion. Ensure the post is removed from the list.

### For the "Scheduled" Tab:
9. **Sorting by Nearest Scheduled Date**
   - **Test:** Verify that posts in the "Scheduled" tab are sorted by the nearest scheduled date.
   - **Scenario:** Confirm that the first post in the list is the one scheduled to publish the soonest, and the last post is the furthest out.

10. **Edit Scheduled Post**
   - **Test:** Ensure the "Edit" option in the dropdown works for scheduled posts.
   - **Scenario:** Select "Edit", make changes, and save. Verify that the post is updated and the scheduled date reflects any changes.

11. **Delete Scheduled Post**
   - **Test:** Verify the "Delete" option in the dropdown works for scheduled posts.
   - **Scenario:** Select "Delete" and confirm the post is removed from the list.

### Additional Edge Case Tests:
12. **Empty States**
    - **Test:** Verify how the UI behaves when there are no drafts, scheduled posts, or published posts.
    - **Scenario:** If there are no posts in a tab, ensure the UI displays an appropriate message like "No posts available."

## Screenshots

my posts page

### PR DESCRIPTION
**Pull request by JohnAllenTech**
# ✨ Codu Pull Request 💻

1/2 closes #1169 

Dont want to put in all in one massive PR. Prefer two bite sized PRs

## Pull Request details

- E2E user one now was a draft and scheduled article
- Testing to ensure that the user can she their scheduled and draft articles on the my-posts page

## Any Breaking changes

- None

## Associated Screenshots

image

image

## [Optional] What gif best describes this PR or how it makes you feel

Alt Text

### CODE REVIEW

File: e2e/setup.ts [hunk 1]
```diff
@@ -19,29 +19,73 @@
     authorId: string,
     commenterId: string,
   ) => {
-    const postId = "1nFnMmN5";
+    const publishedPostId = "1nFnMmN1";
+    const scheduledPostId = "1nFnMmN2";
+    const draftPostId = "1nFnMmN3";
```
[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply e2e/setup.ts]: Tried it. Script complains with 

Error: require() of ES Module /Users/johnallen/oss-repos/codu/node_modules/nanoid/index.js from /Users/johnallen/oss-repos/codu/e2e/setup.ts not supported.
Instead change the require of index.js in /Users/johnallen/oss-repos/codu/e2e/setup.ts to a dynamic import() which is available in all CommonJS modules.


I dont have time to try to troubleshoot it

File: e2e/setup.ts [hunk 2]
```diff
@@ -19,29 +19,73 @@
...
+          slug: "e2e-test-slug-scheduled",
+          likes: 10,
+          readTimeMins: 3,
+          title: "Scheduled Article",
+          body: articleContent,
+          userId: authorId,
+        })
+        .onConflictDoNothing()
+        .returning(),
+    ]);
```
[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply e2e/setup.ts]: Good catch

File: e2e/setup.ts [hunk 3]
```diff
@@ -19,29 +19,73 @@ export const setup = async () => {
...
+          slug: "e2e-test-slug-draft",
+          likes: 10,
+          readTimeMins: 3,
+          title: "Draft Article",
+          body: articleContent,
+          userId: authorId,
+        })
+        .onConflictDoNothing()
+        .returning(),
+
```
[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply e2e/setup.ts]: Good catch!

File: e2e/my-posts.spec.ts [hunk 4]
```diff
@@ -2,14 +1,58 @@
...
 test.describe("Unauthenticated my-posts Page", () => {
-  //
-  // Replace with tests for unauthenticated users
+  test("Unauthenticed users should be redirected to get-started page if they access my-posts directly", async ({
+    page,
+  }) => {
+    await page.goto("http://localhost:3000/my-posts");
+    await page.waitForURL("http://localhost:3000/get-started");
+    expect(page.url()).toEqual("http://localhost:3000/get-started");
+  });
```
[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply e2e/my-posts.spec.ts]: Spelling is hard

File: e2e/setup.ts [hunk 5]
```diff
@@ -19,29 +20,71 @@ export const setup = async () => {
...
-        readTimeMins: 3,
-        title: "Test Article",
-        body: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas vitae ipsum id metus vestibulum rutrum eget a diam. Integer eget vulputate risus, ac convallis nulla. Mauris sed augue nunc. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Nam congue posuere tempor. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Ut ac augue non libero ullamcorper ornare. Ut commodo ligula vitae malesuada maximus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Etiam sagittis justo non justo placerat, a dapibus sapien volutpat. Nullam ullamcorper sodales justo sed.",
-        userId: authorId,
-      })
-      .onConflictDoNothing()
-      .returning();
+
+    const oneYearFromToday = new Date(now);
+    oneYearFromToday.setFullYear(oneYearFromToday.getFullYear() + 1);
```
[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply e2e/setup.ts:33]: Why its only used once?

[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply e2e/setup.ts:33]: I mean its not bad practice but If would do this I would be adding more lines or code then I remove.

File: e2e/setup.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply e2e/setup.ts]: I actually fixed it thanks for pointing it out

**Review by NiallJoeMaher — DISMISSED** [review]

**Comment by NiallJoeMaher** [comment]
@JohnAllenTech a few conflicts

**Comment by JohnAllenTech** [comment]
Should be good to go now @NiallJoeMaher

**Review by NiallJoeMaher — APPROVED** [review]
